lib/taskana-core-test/src/test/java/acceptance/taskcomment/create/CreateTaskCommentAccTest.java

```diff
@@ -0,0 +1,132 @@
+    comment1 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(taskWithComments.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+    comment2 =
+        TaskCommentBuilder.newTaskComment()
+            .taskId(taskWithComments.getId())
+            .textField("Text1")
+            .created(Instant.now())
+            .modified(Instant.now())
+            .buildAndStore(taskService);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_CreateTaskComment_For_TaskThatAlreadyHasComments() throws Exception {
+    TaskComment taskCommentToCreate = taskService.newTaskComment(taskWithComments.getId());
+    taskCommentToCreate.setTextField("Some text");
+    taskService.createTaskComment(taskCommentToCreate);
+
+    List<TaskComment> taskCommentsAfterInsert =
+        taskService.getTaskComments(taskWithComments.getId());
+    assertThat(taskCommentsAfterInsert)
+        .containsExactlyInAnyOrder(comment1, comment2, taskCommentToCreate);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_FailToCreateTaskComment_When_UserHasNoWorkbasketPermission() {
+    TaskComment taskCommentToCreate = taskService.newTaskComment(taskWithComments.getId());
+    taskCommentToCreate.setTextField("Some text");
+
+    assertThatThrownBy(() -> taskService.createTaskComment(taskCommentToCreate))
+        .isInstanceOf(TaskanaException.class)
+        .hasMessageContaining("The current user 'user-1-2' has no '[READ]' permission");
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_CreateTaskComment_When_CopyingAnotherComment() throws Exception {
+    TaskComment taskCommentToCreate = comment1.copy();
+    int sizeBeforeInsertingComment = taskService.getTaskComments(taskWithComments.getId()).size();
+
+    taskService.createTaskComment(taskCommentToCreate);
+
+    List<TaskComment> taskCommentsAfterInsert =
+        taskService.getTaskComments(taskWithComments.getId());
+    assertThat(taskCommentsAfterInsert)
+        .contains(comment1, taskCommentToCreate)
+        .hasSize(sizeBeforeInsertingComment + 1);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @Test
+  void should_FailToCreateTaskComment_When_TaskIdIsNullOrNonExisting() {
+    TaskComment newTaskCommentForNonExistingTask =
+        taskService.newTaskComment("Definetely non existing ID");
+    newTaskCommentForNonExistingTask.setTextField("a newly created taskComment");
+    TaskComment newTaskCommentForTaskIdNull = taskService.newTaskComment(null);
+    newTaskCommentForTaskIdNull.setTextField("a newly created taskComment");
+
+    assertThatThrownBy(() -> taskService.createTaskComment(newTaskCommentForNonExistingTask))
+        .isInstanceOf(TaskNotFoundException.class);
+
+    assertThatThrownBy(() -> taskService.createTaskComment(newTaskCommentForTaskIdNull))
+        .isInstanceOf(TaskNotFoundException.class);
```

Comment:
I think we want to check error messages too. So, we should do it here too. Consider building a `TaskNotFound Exception expectedException` so you just can use equals. See https://github.com/Taskana/taskana/pull/1946/commits/87c8b1ff7a5be74a1493cab0de3cbaabee456938 (after line 647)

Reply:
I will check the text of the error message :)